src/WingetCreateCore/Common/Serialization.cs

```diff
@@ -197,18 +199,61 @@ private static string RemoveBom(string value)
         {
             string bomMarkUtf8 = Encoding.UTF8.GetString(Encoding.UTF8.GetPreamble());
             return value.StartsWith(bomMarkUtf8, StringComparison.OrdinalIgnoreCase) ? value.Remove(0, bomMarkUtf8.Length) : value;
+        }
+
+        /// <summary>
+        /// Custom TypeInspector to priorize properties that have a defined YamlMemberAttribute for custom override.
+        /// </summary>
+        private class AliasTypeInspector : TypeInspectorSkeleton
+        {
+            private readonly ITypeInspector innerTypeDescriptor;
+
+            public AliasTypeInspector(ITypeInspector innerTypeDescriptor)
+            {
+                this.innerTypeDescriptor = innerTypeDescriptor;
+            }
+
+            /// <summary>
+            /// Because certain properties were generated incorrectly, we needed to create custom fields for those properties.
+            /// Therefore to resolve naming conflicts during deserialization, we prioritize fields that have the YamlMemberAttribute defined
+            /// as that attribute indicates an override.
+            /// </summary>
+            public override IEnumerable<IPropertyDescriptor> GetProperties(Type type, object container)
+            {
+                var propertyDescriptors = this.innerTypeDescriptor.GetProperties(type, container);
+                var aliasDefinedProps = type.GetProperties().ToList().Where(p => p.GetCustomAttribute<YamlMemberAttribute>() != null).ToList();

```

Comment:
>aliasDefinedProps [](http://example.com/codeflow?start=20&length=17)

nit: yamlMemberAttributeDefinedProps

Reply:
Added additional check to verify that alias is defined. Therefore I'm keeping the name because it still applies.